nupic/research/spatial_pooler.py

```diff
@@ -1162,16 +1173,37 @@ def _mapColumn(self, index):
     return inputIndex
 
 
-  def _mapPotential(self, index, wrapAround=False):
+  def _getInputIndexesFromSynapses(self, synapses):
     """
-    Maps a column to its input bits. This method encapsulates the topology of
-    the region. It takes the index of the column as an argument and determines
-    what are the indices of the input vector that are located within the
-    column's potential pool. The return value is a list containing the indices
-    of the input bits. The current implementation of the base class only
-    supports a 1 dimensional topology of columns with a 1 dimensional topology
-    of inputs. To extend this class to support 2-D topology you will need to
-    override this method. Examples of the expected output of this method:
+    Returns the indices of the input vector which a given set of potential
+    synapses are connected.
+
+    Parameters:
+    ----------------------------
+    @param synapses:   A set of potential synapses connected to the input
+                       vector.
+    """
+
+    inputIndices = []
+    for synapse in synapses:
+      inputIndex = self.connections.dataForSynapse(synapse).presynapticCell
+      inputIndices.append(inputIndex)
+
+    return inputIndices
+
+
+  def _createSynapsesForColumn(self, index, wrapAround=False):
+    """
+    Maps a column to its input bits, create synapses from the column to these,
+    and initializes the permanences of these synapses. This method encapsulates
+    the topology of the region. It takes the index of the column as an argument
+    and determines what are the indices of the input vector that are located
+    within the column's potential pool. The return value is a list containing
+    the indices of the input bits. The current implementation of the base class
+    only supports a 1 dimensional topology of columns with a 1 dimensional
+    topology of inputs. To extend this class to support 2-D topology you will
```

Comment:
does it mean `Connections` does not support 2D SP columns, while current implementation does?